docs/sources/alerting/set-up/provision-alerting-resources/file-provisioning/index.md

```diff
@@ -759,6 +759,39 @@ deleteMuteTimes:
     name: mti_1
 ```
 
+## Template variable interpolation
+
+Provisioning interpolates environment variables using the `$variable` syntax.
+
+```yaml
+contactPoints:
+  - orgId: 1
+    name: My Contact Email Point
+    receivers:
+      - uid: 1
+        type: email
+        settings:
+          addresses: $EMAIL
+```
+
+In this example, provisioning replaces `$EMAIL` with the value of the `EMAIL` environment variable or an empty string if it is not present. For more information, refer to [Using environment variables in the Provision documentation][provisioning_env_vars].
+
+Note that the provisioning interpolation may unexpectedly substitute template variables in alerting resources.
+
+In alerting resources, you can use template variables with the same `$variable` syntax for some properties. For instance:
+
+- Alert rule labels
+- Alert rule annotations (omits interpolation)
+- Contact point messages
+- Notification template content (omits interpolation)
+
+Properties that omit the provisioning interpolation, like annotation or notification templates, do not require changes. For the others, if you’ve defined a template variable such as `$variable`, **use `$$variable` to avoid interpolation.**
+
+For example, you could use a `subject` property in the `contactPoints.receivers.settings` object alongside an undefined environment variable.
+
+1. `subject: '{{ $labels }}'` will convert to `subject: '{{ }}'` which is incorrect.
+1. `subject: '{{ $$labels }}'` will convert to `subject: '{{ $labels }}'` as desired.
```

Comment:
Here are some suggestions on the wording. I also confirmed all of the fields that do not currently support interpolation so we can comprehensively list them. Feel free to modify/pick and choose what you want from here, though I think listing all of the unsupported fields should at least be included.

```suggestion
In this example, provisioning replaces `$EMAIL` with the value of the `EMAIL` environment variable or an empty string if it is not present. For more information, refer to [Using environment variables in the Provision documentation][provisioning_env_vars].

In alerting resources, most properties support template variable interpolation, with a few exceptions:

- Alert rule annotations: `groups[].rules[].annotations`
- Alert rule time range: `groups[].rules[].relativeTimeRange`
- Alert rule query model: `groups[].rules[].data.model`
- Mute timings name: `muteTimes[].name`
- Mute timings time intervals: `muteTimes[].time_intervals[]`
- Notification template name: `templates[].name`
- Notification template content: `templates[].template`

Note for properties that support interpolation, you may unexpectedly substitute template variables when not intended. To avoid this, you can escape the `$variable` with `$$variable`.

For example, when provisioning a `subject` property in a `contactPoints.receivers.settings` object that is meant to use the `$labels` variable.

1. `subject: '{{ $labels }}'` will interpolate, incorrectly defining the subject as `subject: '{{ }}'`.
1. `subject: '{{ $$labels }}'` will not interpolate, correctly defining the subject as `subject: '{{ $labels }}'`.
```